frontend/src/components/MonacoEditor.tsx

```diff
@@ -133,8 +134,7 @@ const MonacoEditor = ({ content, filename }: Props) => {
         </Button>
       </div>
       <div style={{ fontSize: 14, marginTop: 5, marginBottom: 5 }}>
-        {(!user || !user.me) && 'Please login to enable saving'}
-        {user?.me && currentBranch && `On branch ${currentBranch}`}
+        {user && user.me && commitMessage && `Latest commit message: ${commitMessage}`}
```

Comment:
```suggestion
        {user?.me && commitMessage && `Latest commit: ${commitMessage}`}
```
I think it would be better to remove the "message" part so it doesn't take so much space. The meaning is clear enough without it imo. The check can also be simplified with optional chaining (the way it was earlier).